src/tqec/noise_models/noise_model.py

```diff
@@ -0,0 +1,550 @@
+        ):
+            out_during_moment.append(split_op)
+            return
+
+        args = split_op.gate_args_copy()
+        if self.flip_result:
+            t = OP_TYPES[split_op.name]
+            assert t == MPP or t == JUST_MEASURE_1Q or t == MEASURE_RESET_1Q
+            assert len(args) == 0
+            args = [self.flip_result]
+
+        out_during_moment.append(split_op.name, targets, args)
+        raw_targets = [t.value for t in targets if not t.is_combiner]
+        for op_name, arg in self.after.items():
+            after_moments[(op_name, arg)].append(op_name, raw_targets, arg)
+
+
+class NoiseModel:
+    def __init__(
+        self,
+        idle_depolarization: float,
+        additional_depolarization_waiting_for_m_or_r: float = 0,
+        gate_rules: dict[str, NoiseRule] | None = None,
+        measure_rules: dict[str, NoiseRule] | None = None,
+        any_clifford_1q_rule: NoiseRule | None = None,
+        any_clifford_2q_rule: NoiseRule | None = None,
+    ):
+        self.idle_depolarization = idle_depolarization
+        self.additional_depolarization_waiting_for_m_or_r = (
+            additional_depolarization_waiting_for_m_or_r
+        )
+        self.gate_rules = gate_rules
+        self.measure_rules = measure_rules
+        self.any_clifford_1q_rule = any_clifford_1q_rule
+        self.any_clifford_2q_rule = any_clifford_2q_rule
+
+    @staticmethod
+    def si1000(p: float) -> "NoiseModel":
+        """Superconducting inspired noise.
+
+        As defined in "A Fault-Tolerant Honeycomb Memory":
+        https://arxiv.org/abs/2108.10457
+
+        Small tweak when measurements aren't immediately followed by a reset: the measurement result
+        is probabilistically flipped instead of the input qubit. The input qubit is depolarized after
+        the measurement.
+        """
+        return NoiseModel(
+            idle_depolarization=p / 10,
+            additional_depolarization_waiting_for_m_or_r=2 * p,
+            any_clifford_1q_rule=NoiseRule(after={"DEPOLARIZE1": p / 10}),
+            any_clifford_2q_rule=NoiseRule(after={"DEPOLARIZE2": p}),
+            measure_rules={
+                "Z": NoiseRule(after={"DEPOLARIZE1": p}, flip_result=p * 5),
+            },
+            gate_rules={
+                "R": NoiseRule(after={"X_ERROR": p * 2}),
+            },
+        )
+
+    @staticmethod
+    def uniform_depolarizing(p: float) -> "NoiseModel":
+        """Near-standard circuit depolarizing noise.
+
+        Everything has the same parameter p. Single qubit clifford gates
+        get single qubit depolarization. Two qubit clifford gates get
+        single qubit depolarization. Dissipative gates have their result
+        probabilistically bit flipped (or phase flipped if appropriate).
+
+        Non-demolition measurement is treated a bit unusually in that it
+        is the result that is flipped instead of the input qubit. The
+        input qubit is depolarized.
```

Comment:
```suggestion
        Non-demolition measurement is treated a bit unusually in that it
        is the result that is flipped instead of the input qubit.
```